src/bin/julialauncher.rs

```diff
@@ -159,13 +159,40 @@ fn get_julia_path_from_channel(
     juliaupconfig_path: &Path,
     juliaup_channel_source: JuliaupChannelSource,
 ) -> Result<(PathBuf, Vec<String>)> {
+    let paths = get_paths().with_context(|| "Trying to load all global paths.")?;
+    let versiondb_data =
+        load_versions_db(&paths).with_context(|| "command failed to load versions db.")?;
     let channel_info = config_data
             .installed_channels
             .get(channel)
             .ok_or_else(|| match juliaup_channel_source {
-                JuliaupChannelSource::CmdLine => UserError { msg: format!("ERROR: Invalid Juliaup channel `{}` at command line.", channel).to_string() }.into(),
-                JuliaupChannelSource::EnvVar => UserError { msg: format!("ERROR: Invalid Juliaup channel `{}` in environment variable JULIAUP_CHANNEL.", channel).to_string() }.into(),
-                JuliaupChannelSource::Override => UserError { msg: format!("ERROR: Invalid Juliaup channel `{}` in directory override.", channel).to_string() }.into(),
+                JuliaupChannelSource::CmdLine => {
+                    if versiondb_data.available_channels.contains_key(channel) {
+                        UserError { msg: format!("`{}` is not installed. Please run `juliaup add {}` to install channel or version", channel, channel) }
+                    } else if versiondb_data.available_versions.contains_key(channel) {
```

Comment:
I don't think we need to check in the `availabe_versions`, users can't access those versions in any, I think just looking at `available_channels` is enough. Same applies to the same `if` blocks below.